File: Java/263.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    public boolean isUgly(int num) {
        
        if(num==0)
            return false;
        
        if(num==1)
            return true;
        while(num%2==0)
            num=num/2;
        while(num%3==0)
            num=num/3;
        while(num%5==0)
            num=num/5;
        return num==1;   
    }
}
__________________________________________________________________________________________________
sample 32024 kb submission
class Solution {
    public boolean isUgly(int num) {
		while (num > 1) {
			boolean flag = true;
			if (num % 5 == 0) {
				num /= 5;
				flag = false;
			}
			if (num % 3 == 0) {
				num /= 3;
				flag = false;
			}
			if (num % 2 == 0) {
				num /= 2;
				flag = false;
			}
			if (flag) {
				return false;
			}
		}
		if (num == 1) {
			return true;
		}
		return false;
    }
}
__________________________________________________________________________________________________

File: Python3/263.py
__________________________________________________________________________________________________
sample 20 ms submission
class Solution:
    def isUgly(self, num: int) -> bool:
        if num<=0:
            return False 
        while(num%30==0 ):
            num=num/30
        while(num%10==0 ):
            num=num/10
        
        while(num%5==0 ):
            num=num/5
        while(num%3==0 ):
            num=num/3    
        while(num%2==0 ):
            num=num/2
        if num==1:
            return True
        else:
            return False
__________________________________________________________________________________________________
sample 13068 kb submission
class Solution:
    def isUgly(self, num: int) -> bool:
        if num <= 0:
            return False
        while num & 1 == 0:
            num >>= 1
        while num % 3 == 0:
            num /= 3
        while num % 5 == 0:
            num /= 5
        return num == 1
__________________________________________________________________________________________________
